backend/src/apiserver/resource/resource_manager.go

```diff
@@ -214,6 +217,31 @@ func (r *ResourceManager) CreateRun(apiRun *api.Run) (*model.RunDetail, error) {
 		return nil, util.NewInternalServerError(err, "Failed to create a workflow for (%s)", workflow.Name)
 	}
 
+	hasExperiment := false
```

Comment:
could you move the logic here into a util function, maybe named something like setExperimentIfNotPresent?

Reply:
Done